app/controllers/api/v1/payments_controller.rb

```diff
@@ -50,11 +43,32 @@ def show
         respond_with payments_record
     end
 
+
     private
 
     def payment_params
         params.require(:payment_amount)
     end
+
+    def create_new_payment(sender, receiver, bitcoins_amount, invoice_id)
+        Payment.create!(:sender => sender, 
+                        :receiver => receiver, 
+                        :amount => bitcoins_amount,  
+                        :completed => true, 
+                        :invoice_id => invoice_id)    
+    end
+
+    def receiver_invoice(api_key, api_secret, bitcoins_amount)
+        buda_receiver = BudaUserService.new(api_key: api_key, api_secret: api_secret)
+        invoice = buda_receiver.generate_invoice(bitcoins_amount)
+    end
+
+    def sender_payment(api_key, api_secret, bitcoins_amount, invoice_code)
+        _secrets = Rails.application.secrets
+        buda_payer = BudaUserService.new(api_key: api_key, api_secret: api_secret)
+        invoice_payment = buda_payer.pay_invoice(bitcoins_amount, invoice_code, _secrets.buda_payment_simulation)
```

Comment:
Ya que solo se usa buda_payment_simulation, podrías hacer lo siguiente:
```suggestion
        simulate = Rails.application.secrets.buda_payment_simulation
        buda_payer = BudaUserService.new(api_key: api_key, api_secret: api_secret)
        invoice_payment = buda_payer.pay_invoice(bitcoins_amount, invoice_code, simulate)
```